user-interface/src/main/java/life/qbic/datamanager/views/projects/project/samples/download/SampleInformationXLSXProvider.java

```diff
@@ -0,0 +1,185 @@
+package life.qbic.datamanager.views.projects.project.samples.download;
+
+import static life.qbic.logging.service.LoggerFactory.logger;
+
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Comparator;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import life.qbic.application.commons.ApplicationException;
+import life.qbic.application.commons.ApplicationException.ErrorCode;
+import life.qbic.datamanager.views.general.download.DownloadContentProvider;
+import life.qbic.logging.api.Logger;
+import life.qbic.projectmanagement.application.sample.SamplePreview;
+import life.qbic.projectmanagement.domain.model.experiment.VariableLevel;
+import org.apache.poi.ss.usermodel.Row;
+import org.apache.poi.ss.usermodel.Sheet;
+import org.apache.poi.ss.usermodel.Workbook;
+import org.apache.poi.xssf.usermodel.XSSFWorkbook;
+
+/**
+ * <b>Sample Information XLSX Provider</b>
+ *
+ * <p>Provides the download content as XLSX spreadsheet for the sample information</p>
+ *
+ * @since 1.1.0
+ */
+public class SampleInformationXLSXProvider implements DownloadContentProvider {
+
+  private static final String FILE_NAME = "sample_information.xlsx";
+  private static final Logger log = logger(SampleInformationXLSXProvider.class);
+  private final List<SamplePreview> samples = new ArrayList<>();
+
+
+  private static void setAutoWidth(Sheet sheet) {
+    for (int col = 0; col <= SamplePreviewColumn.values().length; col++) {
+      sheet.autoSizeColumn(col);
+    }
+  }
+
+  private static void formatHeader(Row header, Map<String, Integer> variableNamesToColumn) {
+    for (SamplePreviewColumn value : SamplePreviewColumn.values()) {
+      var cell = header.createCell(value.column());
+      cell.setCellValue(value.header());
+    }
+    for (Map.Entry<String, Integer> entry : variableNamesToColumn.entrySet()) {
+      var cell = header.createCell(entry.getValue());
+      cell.setCellValue(entry.getKey());
+    }
+  }
+
+  public Set<String> uniqueExperimentalVariables(List<SamplePreview> samples) {
+    HashSet<String> variableNames = new HashSet<>();
+    for (SamplePreview sample : samples) {
+      sample.experimentalGroup().condition().getVariableLevels()
+          .forEach(vl -> variableNames.add(vl.variableName().value()));
```

Comment:
You are doing two things in this line. 1) you are mapping all variable levels to their variable names; 2) you are collecting them in a list.
I suggest separating these two steps. You can then use the addAll method to add all to a List.